src/analyses/tutorials/taint.ml

```diff
@@ -0,0 +1,129 @@
+(** An analysis specification for didactic purposes. *)
+(* Helpful link on CIL: https://goblint.in.tum.de/assets/goblint-cil/ *)
+(* Goblint documentation: https://goblint.readthedocs.io/en/latest/ *)
+(* You may test your analysis on our toy examples by running `ruby scripts/update_suite.rb group tutorials` *)
+(* after removing the `SKIP` from the beginning of the tests in tests/regression/99-tutorials/{03-taint_simple.c,04-taint_inter.c} *)
+
+open Prelude.Ana
+open Analyses
+
+module VarinfoSet = SetDomain.Make(CilType.Varinfo)
+
+(* Use to check if a specific function is a sink / source *)
+let is_sink varinfo = Cil.hasAttribute "taint_sink" varinfo.vattr
+let is_source varinfo = Cil.hasAttribute "taint_source" varinfo.vattr
+
+
+(** "Fake" variable to handle returning from a function *)
+let return_varinfo = dummyFunDec.svar
+
+module Spec : Analyses.MCPSpec with module D = Lattice.Unit and module C = Lattice.Unit =
+struct
+  include Analyses.DefaultSpec
+
+  let name () = "taint"
+  module D = Lattice.Unit (* TODO: Change such that you have a fitting local domain; also update in the signature above *)
```

Comment:
Why have it in the signature at all?

Reply:
to have a space that people can easily edit. It's non obvious that one has to overwrite something that is defined somewhere else if you are not familiar with the code base.

Reply:
Not sure what you mean. There's a module constraint on the `Spec` signature above, but it's not really necessary to have that. It can be a fully abstract `MCPSpec`, just like in any analysis, and they wouldn't have to change the definition of `module D` in two places, that's all.

Reply:
True, the entire sharing constraint is unnecessary!

Reply:
Yes, I think it would only be this case, so probably it causes more confusion when it is there than when it is not.